<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from VL.xacro                       | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="VL" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="silver">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <transmission name="waist_q1_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="waist_q1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="waist_q1_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_q2_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_q2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_q2_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="revolution_q3_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="revolution_q3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="revolution_q3_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_q4_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_q4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_q4_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="slider_q5_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="slider_q5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="slider_q5_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_q6_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_q6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_q6_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="control"/>
  </gazebo>
  <gazebo reference="base_link">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
    <gravity>true</gravity>
  </gazebo>
  <gazebo reference="Componente5_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="q1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="q2_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="q3_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="q4_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="q5_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="q6_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="3.019715383457247e-11 3.0286398058123193e-12 0.04300979484933489"/>
      <mass value="4.071597447876389"/>
      <inertia ixx="0.006647" ixy="-0.0" ixz="0.0" iyy="0.00646" iyz="0.0" izz="0.007974"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://vl/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://vl/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Componente5_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-6.551741014804067e-15 0.0 -0.029302995324400033"/>
      <mass value="1.4329758168878506"/>
      <inertia ixx="0.00077" ixy="-0.0" ixz="0.0" iyy="0.0009" iyz="0.0" izz="0.000718"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0 -0.07"/>
      <geometry>
        <mesh filename="package://vl/meshes/Componente5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0 -0.07"/>
      <geometry>
        <mesh filename="package://vl/meshes/Componente5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="q1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.004888834987006494 0.007815175841509137 0.017771527603799395"/>
      <mass value="0.4353288415386871"/>
      <inertia ixx="0.000191" ixy="4.4e-05" ixz="2.6e-05" iyy="0.000139" iyz="-5.1e-05" izz="0.00023"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0 -0.09"/>
      <geometry>
        <mesh filename="package://vl/meshes/q1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0 -0.09"/>
      <geometry>
        <mesh filename="package://vl/meshes/q1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="q2_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.02095807578536191 0.18109836981288438 8.211481883291238e-06"/>
      <mass value="2.5196586015002698"/>
      <inertia ixx="0.036597" ixy="-0.00048" ixz="-0.0" iyy="0.000747" iyz="-2e-06" izz="0.036642"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0045 -0.035737 -0.126223"/>
      <geometry>
        <mesh filename="package://vl/meshes/q2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0045 -0.035737 -0.126223"/>
      <geometry>
        <mesh filename="package://vl/meshes/q2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="q3_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.026292965945851884 0.08156450403034149 3.816233883152664e-06"/>
      <mass value="1.1178385165580391"/>
      <inertia ixx="0.009883" ixy="0.001835" ixz="0.0" iyy="0.000903" iyz="-0.0" izz="0.010502"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0175 -0.425737 -0.12624"/>
      <geometry>
        <mesh filename="package://vl/meshes/q3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0175 -0.425737 -0.12624"/>
      <geometry>
        <mesh filename="package://vl/meshes/q3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="q4_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.02610053974454684 0.19019012678385883 7.778900701443936e-06"/>
      <mass value="2.3634814905972625"/>
      <inertia ixx="0.017319" ixy="0.00018" ixz="0.0" iyy="0.002243" iyz="-1e-06" izz="0.017344"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0555 -0.569737 -0.126247"/>
      <geometry>
        <mesh filename="package://vl/meshes/q4_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0555 -0.569737 -0.126247"/>
      <geometry>
        <mesh filename="package://vl/meshes/q4_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="q5_1">
    <inertial>
      <origin rpy="0 0 0" xyz="2.9004576518332215e-15 0.1604149425810586 7.175376755985674e-06"/>
      <mass value="1.8115927533864018"/>
      <inertia ixx="0.012478" ixy="-0.0" ixz="-0.0" iyy="0.001143" iyz="-0.0" izz="0.012478"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.082 -0.654737 -0.12625"/>
      <geometry>
        <mesh filename="package://vl/meshes/q5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.082 -0.654737 -0.12625"/>
      <geometry>
        <mesh filename="package://vl/meshes/q5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="q6_1">
    <inertial>
      <origin rpy="0 0 0" xyz="4.3576253716537394e-15 0.042295913006092034 1.7662381922300963e-06"/>
      <mass value="1.5169719326279612"/>
      <inertia ixx="0.001215" ixy="0.0" ixz="0.0" iyy="0.001083" iyz="-0.0" izz="0.001291"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.082 -0.924737 -0.126262"/>
      <geometry>
        <mesh filename="package://vl/meshes/q6_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.082 -0.924737 -0.126262"/>
      <geometry>
        <mesh filename="package://vl/meshes/q6_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="Rigida" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.07"/>
    <parent link="base_link"/>
    <child link="Componente5_1"/>
  </joint>
  <joint name="waist_q1" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.02"/>
    <parent link="Componente5_1"/>
    <child link="q1_1"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="100" lower="-3.141593" upper="3.141593" velocity="100"/>
  </joint>
  <joint name="shoulder_q2" type="revolute">
    <origin rpy="0 0 0" xyz="-0.0045 0.035737 0.036223"/>
    <parent link="q1_1"/>
    <child link="q2_1"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="0.0" upper="3.141593" velocity="100"/>
  </joint>
  <joint name="revolution_q3" type="revolute">
    <origin rpy="0 0 0" xyz="0.022 0.39 1.7e-05"/>
    <parent link="q2_1"/>
    <child link="q3_1"/>
    <axis xyz="-0.0 -1.0 -4.3e-05"/>
    <limit effort="100" lower="-3.141593" upper="3.141593" velocity="100"/>
  </joint>
  <joint name="elbow_q4" type="revolute">
    <origin rpy="0 0 0" xyz="-0.073 0.144 7e-06"/>
    <parent link="q3_1"/>
    <child link="q4_1"/>
    <axis xyz="-1.0 0.0 -0.0"/>
    <limit effort="100" lower="-3.141593" upper="3.141593" velocity="100"/>
  </joint>
  <joint name="slider_q5" type="prismatic">
    <origin rpy="0 0 0" xyz="-0.0265 0.085 3e-06"/>
    <parent link="q4_1"/>
    <child link="q5_1"/>
    <axis xyz="0.0 1.0 4.3e-05"/>
    <limit effort="100" lower="0.00" upper="0.24" velocity="100"/>
  </joint>
  <joint name="wrist_q6" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.27 1.2e-05"/>
    <parent link="q5_1"/>
    <child link="q6_1"/>
    <axis xyz="-0.0 -1.0 -4.3e-05"/>
    <limit effort="100" lower="-3.141593" upper="3.141593" velocity="100"/>
  </joint>
</robot>

